drivers/adc/adc_sam0.c

```diff
@@ -392,7 +375,20 @@ static int start_read(const struct device *dev,
 
 	wait_synchronization(adc);
 
-	if (sequence->channels != 1U) {
+	active_channels = 0;
+	for (i = 0; i < 32; i += 1) {
+		if (sequence->channels & BIT(i)) {
+			active_channels += 1;
+		}
+	}
+	if (active_channels != 1U) {
```

Comment:
I'll suggest replacing your bit check with this (more elegant and faster) (See PR 

```
-	if (sequence->channels != 1U) {
+	if ((sequence->channels == 0) || 
+		((sequence->channels & (sequence->channels - 1)) != 0)) {
```

Then you can also remove the two added variables (i and active_channels) defined 50 lines above

Reply:
Thanks! It took me a sec to figure out what is going on, but it looks good.